lib/Dialect/Ibis/IbisOps.cpp

```diff
@@ -34,6 +35,48 @@ void printScopeRefFromName(OpAsmPrinter &p, Operation *op, Type type,
   // symbol.
 }
 
+// Returns a name for Ibis values.
+// NOLINTBEGIN(misc-no-recursion)
+static StringAttr genValueName(Value scoperef) {
```

Comment:
Since this is recursive, would it be wise to return something which isn't immortal then convert it to a stringattr? Ideally, you'd pass a string_ostream into the function. Just a thought.

Reply:
Fair point, modified to be based on a sstream